db/app.js

```diff
@@ -12,9 +13,33 @@ app.get('/api', (req, res) => {
 
 app.get("/api/topics", getTopics);
 
+app.get('/api/articles/:article_id', getArticle)
+
+
 app.all("*",(req, res) => {
-  res.status(500).send({ msg: "Invalid Path!" });
+  res.status(404).send({ msg: "Invalid Path!" });
 });
 
+app.use((err, req, res, next) => {
+  if (err.code === '22P02') {
+    res.status(400).send({msg: 'Bad request!'})
+  } else {
+    next(err)
+  }
+})
+
+app.use((err, req, res, next) => {
+
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg })
+  } else {
+    next(err)
+  }
+})
+
```

Comment:
Good error handling here! I would however advise putting this into an error handler file and then importing in the functions to app.js. The error handlers should be split by functionality (eg handleCustomErrors, handlePsqlErrors). It may just tidy up the code a bit once you get more and more errors :)